src/cheri_insts.sail

```diff
@@ -1741,22 +1741,17 @@ function handle_store_cond_cap_via_cap(cs2, auth_idx, auth_val, vaddrBits) = {
       TR_Address(addr, _) => {
         let eares : MemoryOpResult(unit) = mem_write_ea_cap(addr, aq, rl, false);
         match (eares) {
-          MemException(e) => { handle_mem_exception(addr, e); RETIRE_FAIL },
+          MemException(e) => { handle_mem_exception(vaddrBits, e); RETIRE_FAIL },
           MemValue(_) => {
             let res : MemoryOpResult(bool) = mem_write_cap(addr, cs2_val, aq, rl, false);
             match (res) {
-              MemValue(true)  => {
-                 C(cs2) = int_to_cap(EXTZ(0b0));
-                 cancel_reservation();
-                 RETIRE_SUCCESS
-              },
-              MemValue(false) => {
-                 C(cs2) = int_to_cap(EXTZ(0b1));
+              MemValue(success)  => {
+                 C(cs2) = int_to_cap(EXTZ(bool_to_bits(not (success))));
```

Comment:
```suggestion
              MemValue(success) => {
                 C(cs2) = int_to_cap(EXTZ(bool_to_bits(not(success))));
```

We're inconsistent with `not`, but I think this looks better here (and given it's really a function perhaps should be how we always format it).